packages/aft/lib/src/commands/docs_command.dart

```diff
@@ -0,0 +1,389 @@
+      'port',
+      abbr: 'p',
+      defaultsTo: '0',
+      help: 'The port to serve on',
+    );
+  }
+
+  late final port = int.parse(argResults!['port'] as String);
+
+  @override
+  String get name => 'serve';
+
+  @override
+  String get description => 'Serves API documentation locally, automatically '
+      'rebuilding when files have changed';
+
+  @override
+  Future<void> run() async {
+    await super.run();
+    await linkPackages();
+
+    // Run initial build
+    final outputDir = p.join(rootDir.path, 'doc/api');
+    await _buildAllDocs(outputDir);
+
+    // Set up filesystem watchers
+    final watchers = <_PackageWatcher>[];
+    for (final package in commandPackages.values) {
+      watchers.add(
+        _PackageWatcher(
+          package,
+          onChange: () async {
+            logger.info('Rebuilding docs for ${package.name}...');
+            try {
+              await _buildDocs(package, outputDir);
+              logger.info('Rebuilt docs for ${package.name}.');
+            } on Exception catch (e) {
+              logger.error('Error rebuilding docs for ${package.name}:', e);
+            }
+          },
+        ),
+      );
+    }
+
+    // Serve output HTML on webserver
+    final server = await io.serve(
+      createStaticHandler(outputDir, defaultDocument: 'index.html'),
+      InternetAddress.anyIPv4,
+      port,
+    );
+    logger
+      ..info('Serving docs on http://localhost:${server.port}')
+      ..info('Press Ctrl+C to stop.');
+
+    // Wait for interruption
+    await StreamGroup.merge([
+      ProcessSignal.sigint.watch(),
+      ProcessSignal.sigterm.watch(),
+    ]).first;
+
+    // Shutdown
+    logger.debug('Shutting down...');
+    await Future.wait(watchers.map((watcher) => watcher.close()));
+    await server.close(force: true);
+  }
+}
+
+/// Watches a [package] for changes and calls [onChange] whenever a file is changed.
+class _PackageWatcher with Closeable {
+  _PackageWatcher(this.package, {required this.onChange}) {
+    _listenForEvents();
+  }
+
+  final PackageInfo package;
+  final Future<void> Function() onChange;
+  late final StreamSubscription<FileSystemEvent> _subscription;
+
+  void _listenForEvents() {
+    final directoriesToWatch = [
+      for (final subpath in const ['doc/lib', 'lib'])
+        Directory(p.join(package.path, subpath)),
+    ].where((dir) => dir.existsSync());
+    final eventStream = StreamGroup.merge([
+      for (final directory in directoriesToWatch)
+        directory.watch(
+          events: FileSystemEvent.all,
+          recursive: true,
+        ),
+    ]);
+    Future<void>? currentOperation;
+    _subscription = eventStream.listen((_) {
+      currentOperation ??= onChange().whenComplete(() {
+        currentOperation = null;
+      });
+    });
+  }
+
+  @override
+  Future<void> close() => _subscription.cancel();
+}
+
+/// Index page template to make navigation between package docs easier.
+const _indexTmpl = r'''
```

Comment:
Once this is built as a binary, locating an HTML file becomes difficult. Embedding the file in the binary would still need to happen, but since there are not really any good options for this in the Dart community, this is the best way right now.